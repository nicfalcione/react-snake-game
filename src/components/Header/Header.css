.game-over {
    width: 800px;
    height: 64px;
    min-width: 800px;
    min-height: 64px;
    margin: auto;
    border: 5px solid black;
    background-color: black;
}

.game-score-wrapper {
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.game-header-name {
    font-size: 3rem;
    color: white;
}