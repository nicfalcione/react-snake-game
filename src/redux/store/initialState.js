import { RIGHT } from '../../constants/constants';

const defaultState = {
    isRunning: true,
    gameOver: false,
    direction: RIGHT,
    score: 0,
    speed: 100,
    snakeSegments:[
        {
            key: 1,
            xPos: 380,
            yPos: 220
        },
        {
            key: 2,
            xPos: 360,
            yPos: 220
        },
        {
            key: 3,
            xPos: 340,
            yPos: 220
        }
    ],
    apple: {
        xPos: 400,
        yPos: 400
    }
};

export default defaultState;