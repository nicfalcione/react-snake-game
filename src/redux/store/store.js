import { createStore } from '@reduxjs/toolkit';
import { gameStateReducer } from '../reducers/gameStateReducer';

const store = createStore(gameStateReducer);

export default store;