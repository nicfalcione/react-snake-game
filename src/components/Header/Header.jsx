import React from "react";
import { useSelector } from "react-redux";
import Score from '../Score/Score';
import './Header.css';

const Header = () => {
    const score = useSelector((state) => state.score);

    return (
        <div className="game-header">
            <div className="game-score-wrapper">
                <span className="game-over-name">Snake</span>
                <Score score={ score }></Score>
            </div>
        </div>
    );
}

export default Header;