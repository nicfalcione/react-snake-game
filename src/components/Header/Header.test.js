import { render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import store from '../../redux/store/store';
import Header from './Header';

const setupRender = () => {
    return render(
        <Provider store={store}><Header/></Provider>
    )
}

describe('Header component tests', () => {
    it('should render the Game name in the header', () => {
        const { getByText } = setupRender();
        expect(getByText('Snake')).toBeInTheDocument();
        expect(getByText('Score: 0')).toBeInTheDocument();
    });

    it('should match the snapshot', () => {
        const { container } = setupRender();
        expect(container).toMatchSnapshot();
    });
});