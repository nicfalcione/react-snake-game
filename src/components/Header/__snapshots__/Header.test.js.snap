// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests should match the snapshot 1`] = `
<div>
  <div
    class="game-header"
  >
    <div
      class="game-score-wrapper"
    >
      <span
        class="game-over-name"
      >
        Snake
      </span>
      <span
        class="game-score"
      >
        Score: 0
      </span>
    </div>
  </div>
</div>
`;
